src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -277,13 +277,19 @@ protected virtual bool IsAttributeConditionallyOmitted(NamedTypeSymbol attribute
                 ArrayBuilder<BoundExpression> boundNamedArgumentsBuilder = null;
                 HashSet<string> boundNamedArgumentsSet = null;
 
+                // VariableUsePass assumes we're not in a scope that can reference local variables.
+                // Need to rework VariableUsePass if that ever changes.
+                Debug.Assert((this.ContainingMemberOrLambda as MethodSymbol)?.MethodKind != MethodKind.LocalFunction);
+
                 // Avoid "cascading" errors for constructor arguments.
                 // We will still generate errors for each duplicate named attribute argument,
                 // matching Dev10 compiler behavior.
                 bool hadError = false;
 
                 var shouldHaveName = false;
 
+                var variableUsePass = this.IsSemanticModelBinder ? null : new VariableUsePass(this.Compilation.SourceAssembly);
```

Comment:
>new VariableUsePass(this.Compilation.SourceAssembly) [](start = 74, length = 52)

We probably should avoid allocating VariableUsePass when there are no arguments.